src/test/java/com/github/lordrex34/config/TestConfigArray.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2017 Reginald Ravenhorst <lordrex34@gmail.com>
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.github.lordrex34.config;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertThat;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.github.lordrex34.config.impl.ConfigArrayTest;
+import com.github.lordrex34.config.impl.IConfigMarker;
+
+/**
+ * @author lord_rex
+ */
+public class TestConfigArray
```

Comment:
(why did you separate such tests into so many classes? if you would have user @Before instead of @BeforeClass many of them could be inside the very same class as the initialization is the same)

Reply:
(as a general rule you should VERY rarely use @beforeclass)

Reply:
New approach added - please re-check.